src/object_store/src/object/mod.rs

```diff
@@ -399,10 +404,13 @@ impl MonitoredStreamingReader {
             .with_label_values(&[self.media_type, operation_type])
             .start_timer();
         self.operation_size += data_len;
-        let ret =
+        let ret = tokio::time::timeout(self.streaming_read_bytes_timeout, async {
             self.inner.read_exact(buf).await.map_err(|err| {
                 ObjectError::internal(format!("read_bytes failed, error: {:?}", err))
-            });
+            })
+        })
+        .await
+        .unwrap_or_else(|_| Err(ObjectError::internal("read_bytes timeout")));
```

Comment:
Can we expose the `streaming_read_bytes_timeout` directly in Error ?,so it's easier to find the problem?

Reply:
ObjectErrorInner is classified by underlying impl + an Internal. I think it's weird to add another error kind.
```
enum ObjectErrorInner {
    #[error(transparent)]
    S3(BoxedError),
    #[error("disk error: {msg}")]
    Disk {
        msg: String,
        #[source]
        inner: io::Error,
    },
    #[error(transparent)]
    Opendal(opendal::Error),
    #[error(transparent)]
    Mem(crate::object::mem::Error),
    #[error("Internal error: {0}")]
    Internal(String),
}
```